salt/utils/event.py

```diff
@@ -514,7 +516,10 @@ def destroy(self, linger=5000):
             # Wait at most 2.5 secs to send any remaining messages in the
             # socket or the context.term() below will hang indefinitely.
             # See https://github.com/zeromq/pyzmq/issues/102
-            self.sub.setsockopt(zmq.LINGER, linger)
+            try:
```

Comment:
The reason is actually quite simple-- the problem is that __del__ (on interpreter shutdown) is not gauranteed to run in order. I've seen this before-- basically the module zmq was unloaded before this function was run-- so zmq.LINGER isn't valid-- and you get errors. A better way to do this is to just try/except in __del__ and ignore it (so you don't get the error on stderr).